src/ucp/proto/proto_select.c

```diff
@@ -16,27 +16,31 @@
 #include <ucp/core/ucp_context.h>
 #include <ucp/dt/dt.h>
 #include <ucs/datastruct/array.h>
+#include <ucs/datastruct/dynamic_bitmap.h>
 
 #include <ucp/core/ucp_worker.inl>
 
 
-/* Parameters structure for initializing protocols for a selection parameter */
 typedef struct {
-    const ucp_proto_select_param_t *select_param; /* Protocol selection parameter */
-    ucp_proto_id_mask_t            mask;          /* Which protocols are valid */
-    ucp_proto_caps_t               caps[UCP_PROTO_MAX_COUNT]; /* Protocols capabilities */
-    void                           *priv_buf;     /* Protocols configuration buffer */
-    size_t                         priv_offsets[UCP_PROTO_MAX_COUNT]; /* Offset of each
-                                                                         protocol's private
-                                                                         area in 'priv_buf' */
-} ucp_proto_select_init_protocols_t;
+    ucp_proto_id_t   proto_id;
+    size_t           priv_offset;
+    size_t           cfg_thresh; /* Configured protocol threshold */
+    unsigned         cfg_priority; /* Priority of configuration */
```

Comment:
Can we merge these fields with `ucp_proto_caps_t caps` fields?

Reply:
will remove these fields from proto_caps in next PR (after updating existing protocols to use probe)